test/algorithms/set_operations/difference/difference.cpp

```diff
@@ -22,15 +22,42 @@
 
 #include <boost/geometry/io/wkt/wkt.hpp>
 
+#include <boost/geometry/util/constexpr.hpp>
+
 #include <boost/geometry/geometries/point_xy.hpp>
 
 #include "test_difference.hpp"
 #include <algorithms/test_overlay.hpp>
 #include <algorithms/overlay/overlay_cases.hpp>
 #include <algorithms/overlay/multi_overlay_cases.hpp>
 
+namespace
+{
+
+// Change compiler defines to constexpr bools
+// to make conditions more readable
+// and to always compile all code.  
```

Comment:
This is done in `difference.cpp` only, until now.
If you agree, I'll apply it in the others too. And we can apply it everywhere.
Advantages:
* `if` i/o `ifdef`
* code is always compiled
* easier to read

Also, we can harmonize it with the removing of the rescaling, as done in this source file.